### ISSUE TITLE
Aggregate SOQL support.

### ISSUE THREAD

**Issue by Vishal-skywalker** [apex-enterprise-patterns/fflib-apex-common#466]
**The Query factory does not have Aggregate( GROUP BY ) in it.**
We use aggregate queries quite often in terms of reporting and metrics as of now to my knowledge fflib queryFactory does not support such queries, so we have to write the queries manually which is fine but the Selector class does not look good. So thought I could add the said feature, if its ok. 

**Solution you'd like**
I would like to add some methods in the fflib_QueryFactory class to support the above feature.

Already started working on it just for fun, let me know if its something that can be added to the library.

PS:- Thanks for the awesome library.

**Comment by daveespo** [comment]
Hi @Vishal-skywalker  -- thanks for the compliment! Support for Aggregate SOQL is not something we will include in fflib-apex-common. `fflib_SObjectSelector` and `fflib_QueryFactory` are both built around querying for SObject record instances. 

Specifically:
 * The legacy CRUD and FLS functionality presumes the selected field list are concrete SObjectFields
 * Support for subqueries is part of QueryFactory and they are not compatible with Aggregate SOQL
 * For SObjectSelector, the FIELD_LIST is the 'minimum breath' of the SObject record you expect to retrieve -- this is likely too broad for most GROUP BY clauses given that there are limitations on the field types that can be used in GROUP BY

In short, bending QueryFactory and SObjectSelector to support Aggregate SOQL would create confusion on what's supported and what isn't.

**Comment by Vishal-skywalker** [comment]
Sure
